lewis_emulators/jsco4180/interfaces/stream_interface.py

```diff
@@ -24,13 +26,6 @@ def __init__(self):
             CmdBuilder(self.get_pressure_max).escape("a_pmax load p").eos().build(),
             CmdBuilder(self.set_pressure_min).int().escape(" pmin set").build(),
             CmdBuilder(self.get_pressure_min).escape("a_pmin load p").eos().build(),
-            CmdBuilder(self.set_pump_off_timer).float().escape(" offtimer set").build(),
-            CmdBuilder(self.set_pump_on_timer).float().escape(" ontimer set").build(),
-            CmdBuilder(self.set_valve_position).int().escape(" valve set").build(),
-            CmdBuilder(self.get_valve_position).escape("valve load p").eos().build(),
-            CmdBuilder(self.set_file_number).int().escape(" fileno set").build(),
```

Comment:
Rather than remove the file opening stuff from the emulator we should emulate it properly such that the emulator will behave like the device if  you try to set a component without having a file open.

Reply:
You're correct. And the Jasco rep confirmed some issues related to file open/close/setting. I have re-implemented and included logic for pump crash behaviour related to file processing.